src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,134 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double STEP_UPHILL = 1.5d;
+    private Object[] dataCollection;
+    private int size;
+
+    public ArrayList() {
+        dataCollection = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int number) {
+        if (number > 0) {
+            dataCollection = new Object[number];
+        } else {
+            throw new IllegalArgumentException("Illegal argument: " + number);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == dataCollection.length) {
+            dataCollection = grow(size + 1);
+        }
+        dataCollection[size] = value;
+        size = size + 1;
```

Comment:
it is better to use an increment
```suggestion
        size++;
```